packages/checkout/widgets-lib/src/widgets/sale/hooks/useSignOrder.ts

```diff
@@ -205,13 +203,15 @@ export const useSignOrder = (input: SignOrderInput) => {
           await txnResponse?.wait();
         }
 
-        transactionHash = txnResponse?.hash || '';
+        const transactionHash = txnResponse?.hash;
+        if (!transactionHash) {
+          throw new Error('Transaction hash is undefined');
```

Comment:
As this new Error is thrown is it worth capturing that case within the reason below in the try/catch with some more detail around missing txn hash?
Otherwise would it just be showing up as wallet_failed error for default, maybe thats enough?

Reply:
Thanks @dreamoftrees. 
I'm not sure why `transactionHash` can be missing.
Before it was just being returned and if it was `undefined` the parent function would set an empty error (see https://github.com/immutable/ts-immutable-sdk/pull/1878/files#r1634210544), so I refactored it here so at least we have the generic error messaging if it does happen.